main/src/ca/uwaterloo/flix/util/StringCursor.scala

```diff
@@ -0,0 +1,124 @@
+package ca.uwaterloo.flix.util
+
+/**
+  * A class to iterate through an array of characters while maintaining the line and column index
+  * of the cursor.
+  */
+class StringCursor(val data: Array[Char]) {
```

Comment:
Move this into the end of the Lexer class, make it private, and rename it to ArrayCursor?

Reply:
You don't want a separate file? Isn't that nicer?

Reply:
No because I don't want it reused outside the Lexer.

Reply:
Okay. I think StringCursor is a better name than ArrayCursor since data is interpreted as a String (I'm also not sure why Source doesn't just have a String instead of converting it into an array)